File: c/LeetCode_367_ValidPerfectSquare.c
#include <stdio.h>
#include <stdlib.h>


typedef enum {false, true} bool;


bool isPerfectSquare(int num) {
    if (num == 1 || num == 0) {
        return 1;
    }

    int low = 0;
    int mid = 0;
    int high = num; 
    while (low < high - 1) {
        mid = low + ((high - low) / 2);

        int sqr = mid * mid;
        if (sqr == num) {
            return 1;
        } else if (sqr > num) {
            high = mid;
        } else if (sqr < num) {
            low = mid;
        }
    }

    return 0;
}


int main() {
    return 0;
}

File: python/LeetCode_367_ValidPerfectSquare.py
def isPerfectSquare(num: int) -> bool:
    low = 0 
    high = num 
    while (low < high - 1):
        mid = low + ((high - low) // 2)
        
        if (mid ** 2 == num):
            return 1
        
        if (mid ** 2 > num):
            high = mid 
        elif (mid ** 2 < num):
            low = mid 

    return 0 


print(isPerfectSquare(169))